src/acorn/src/expression.js

```diff
@@ -214,7 +214,12 @@ pp.parseExprSubscripts = function(refShorthandDefaultPos) {
 }
 
 pp.parseSubscripts = function(base, start, noCalls) {
-  if (this.eat(tt.dot)) {
+  if (this.eat(tt.doubleColon)) {
```

Comment:
Thanks Kevin!

As for the first case - makes sense to me.

As for the second one - I don't see any benefits from such construct at all since second `.bind` anyway doesn't change context of prebound function, so it's kinda convoluted example IMO.